src/fetching/provider/query_service.rs

```diff
@@ -67,7 +67,9 @@ where
         match res {
             Ok((payload, common)) => {
                 // Verify that the data we retrieved is consistent with the request we made.
-                let num_storage_nodes = VidSchemeType::get_num_storage_nodes(common.common());
+                let num_storage_nodes =
+                    usize::try_from(VidSchemeType::get_num_storage_nodes(common.common()))
```

Comment:
I think we can just do `as u32` here. IMO we will only run on 32 bit systems or larger and stuff like this just makes it harder to read thecode